tests/writers/test_s3_writer.py

```diff
@@ -0,0 +1,60 @@
+import boto3
+import json
+
+from moto import mock_s3
+from unittest import TestCase
+from parameterized import parameterized
+from nck.writers.s3_writer import S3Writer
+from nck.streams.json_stream import JSONStream
+
+
+list_dict = [{"a": "4", "b": "5", "c": "6"}, {"a": "7", "b": "8", "c": "9"}]
+
+
+def dict_generator(list_dict):
+    for di in list_dict:
+        yield di
+
+
+def mock_stream(list_dict, name):
+    return JSONStream(name, dict_generator(list_dict))
+
+
+@mock_s3
+class S3WriterTest(TestCase):
+    @classmethod
+    @mock_s3
+    def setUpClass(cls):
+        client1 = boto3.resource("s3", region_name="us-east-1")
+        client1.create_bucket(Bucket="test")
+
+    def test_bucket_doesnt_exist(self):
+        with self.assertRaisesRegex(Exception, ".bucket does not exist."):
```

Comment:
Maybe we could be more precise and test that we have the bucket in the error message like this:

```suggestion
        with self.assertRaisesRegex(Exception, "non-existing-bucket bucket does not exist."):
```

Also, I guess we can't verify that it logs the list of available buckets?